packages/docs/docs/docs/testing.md

```diff
@@ -3,85 +3,152 @@ id: testing
+    result.should.eq('A sample');
   })
 
   it('should create a proxy for the EVM package', async function () {
     const proxy = await this.project.createProxy(ERC20, { contractName: 'StandaloneERC20', packageName: 'openzeppelin-eth' });
-    const result = await proxy.totalSupply();
-    result.toNumber().should.eq(0);
+    const result = await proxy.methods.totalSupply().call();
+    result.should.eq('0');
   })
 })
+
+```
+
+Next, modify your `package.json` file to include the following script:
+
+```json
+"test": "NODE_ENV=test truffle test"
+```
+
+And run the test in your console with:
+
+```console
+npm test
+```
+
+That's it! Now, lets look at what we just did in more detail.
+
+## Understanding the test script
+
+We first import `TestHelper` from `zos`. This helper facilitates the creation of a project object that will set up the entire ZeppelinOS project within a test environment.
+
+```js
+import { TestHelper } from 'zos';
+```
+
+We are also importing `Contracts` and `ZWeb3` from `zos-lib`. `Contracts` helps us retrieve compiled contract artifacts, while `ZWeb3` is needed to set up our Web3 provider to ZeppelinOS.
+
+```js
+import { Contracts, ZWeb3 } from 'zos-lib';
+
+ZWeb3.initialize(web3.currentProvider);
+
+const Sample = Contracts.getFromLocal('Sample');
+const ERC20 = Contracts.getFromNodeModules('openzeppelin-eth', 'ERC20');
+```
+
+We then invoke `TestHelper` in the test, optionally including a set of options to be used when deploying the contracts (such as `from`, `gas`, and `gasPrice`):
+```js
+beforeEach(async function () {
+  this.project = await TestHelper({ from: owner })
+});
+```
+
+And finally, we add the tests themselves. Notice how each test first creates a proxy for each contract:
+
+```js
+const proxy = await this.project.createProxy(Sample);
+```
+
+The [createProxy](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/BaseSimpleProject.ts#L96) method of the project accepts an additional object parameter in which you can specify an initializer function with arguments, just as you would by using the regular `zos create` command from the CLI, but due to the simplicity of this example, it's not necessary in our case.
+
+Also, notice that the way we interact with the contracts is by using their `methods` object. This is because ZeppelinOS uses Web3 1.0 Contract interface:
+
+```js
+const result = await proxy.methods.totalSupply().call();
 ```
 
-Use this convenient tool to write tests for your code and storage migrations before executing them in production.
+This is how you should write tests for your ZeppelinOS projects. The project object provided by `TestHelper` wraps all of ZeppelinOS programmatic interface in a way that is very convenient to use in tests. By running tests in this way, you make sure that you are testing your contracts with the exact set of conditions that they would have in production, after you deploy them with ZeppelinOS.
```

Comment:
I disagree here. The article had no narrative before these changes, and seemed too obscure for someone that is new to the concepts it introduces.  It needs to be super clear to the user that this type of testing differs from regular testing in that it uses upgradeable instances instead of logic contracts. It's not just a matter of style that could be added by a writer, it is a technical concept that is being explained. It might be boring if you already know this stuff, but not if it's new to you. On the contrary, being too concise would bore readers because they don't understand or don't know why they are doing the tests like this anyway.

Reply:
> On the contrary, being too concise would bore readers because they don't understand or don't know why they are doing the tests like this anyway.

I don't mean to remove explanations, just to go with a more succinct style altogether. I think we agree in that the story needs to be complete, the argument is more of style - but I'm fine to disagree and ~commit~ approve :-)

Reply:
The detailed explanations were extremely helpful for me: I skimmed through it first and implemented. Then, whenever I encountered a problem I came back to it and found exactly the information I needed :100: 
The only thing that I thought was missing that would actually be helpful is a pointer on how to pass the initArgs (since it will always be needed for a serious project).

> The [createProxy](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/BaseSimpleProject.ts#L96) method of the project accepts an additional object parameter in which you can specify an initializer function with arguments, just as you would by using the regular `zos create` command from the CLI

This is not entirely correct or at least a bit confusing, since the parameters here have to get passed into an array value of the initArgs object property.

Reply:
@ajsantander do you think you can add a note on the initArgs, as @acrdlph suggests?

Reply:
@acrdlph sure! I expanded the explanation a bit on how to call the project's createProxy method with an initializer function call.

Just FYI, the article will show that you do so like this:
```js
const proxy = await this.project.createProxy(Sample, {
  initMethod: 'initialize',
  initArts: [42]
});
```